geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java

```diff
@@ -187,19 +190,289 @@ public static boolean hasRasterLiteSupport(){
+     * @param coverageName the table to use.
+     * @param width        of image in pixel.
+     * @param height       of image in pixel.
+     * @param tileBounds   [west,south,east,north] [minx, miny, maxx, maxy] bounds.
+     * @param styleName    used in coverage. default: 'default'
+     * @param mimeType     'image/tiff' etc. default: 'image/png'
+     * @param bgColor      html-syntax etc. default: '#ffffff'
+     * @param transparent  0 to 100 (?).
+     * @param quality      0-100 (for 'image/jpeg')
+     * @param reaspect     1 = adapt image width,height if needed based on given bounds
+     * @return the image data as byte[]
+     */
+     // GetMapImageFromVector(text coverage, BLOB geom, int width, int height,text style, text format, text bg_color,	int transparent, int quality, int reaspect)             
+    public static byte[] rl2_GetMapImageFromVector(Database dbSpatialite, String sourceSrid, String destSrid, String coverageName, int width,
+                                         int height, double[] tileBounds, String styleNameVector, String mimeType, String bgColor, int transparent, int quality,
+                                         int reaspect) {
+        boolean doTransform = false;
+        if (!sourceSrid.equals(destSrid)) {
+            doTransform = true;
+        }
+        // sanity checks
+        if (styleNameVector.equals(""))
+            styleNameVector = "default";
+        if (mimeType.equals(""))
+            mimeType = "image/png";
+        if (bgColor.equals(""))
+            bgColor = "#ffffff";
+        if ((transparent < 0) || (transparent > 100))
+            transparent = 0;
+        if ((quality < 0) || (quality > 100))
+            quality = 0;
+        if ((reaspect < 0) || (reaspect > 1))
+            reaspect = 1; // adapt image width,height if needed based on given bounds [needed for
+        // tiles]
+        StringBuilder mbrSb = new StringBuilder();
+        if (doTransform)
+            mbrSb.append("ST_Transform(");
+        mbrSb.append("BuildMBR(");
+        mbrSb.append(tileBounds[0]);
+        mbrSb.append(",");
+        mbrSb.append(tileBounds[1]);
+        mbrSb.append(",");
+        mbrSb.append(tileBounds[2]);
+        mbrSb.append(",");
+        mbrSb.append(tileBounds[3]);
+        if (doTransform) {
+            mbrSb.append(",");
+            mbrSb.append(sourceSrid);
+            mbrSb.append("),");
+            mbrSb.append(destSrid);
+        }
+        mbrSb.append(")");
+        // SELECT
+        // RL2_GetMapImageFromVector('1890.berlin_postgrenzen',BuildMBR(20800.0,22000.0,24000.0,19600.0),1200,1920,'default','image/png','#ffffff',0,0,1);
+        String mbr = mbrSb.toString();
+        StringBuilder qSb = new StringBuilder();
+        qSb.append("SELECT RL2_GetMapImageFromVector('");
+        qSb.append(coverageName);
+        qSb.append("',");
+        qSb.append(mbr);
+        qSb.append(",");
+        qSb.append(Integer.toString(width));
```

Comment:
on the stringbuilder you do not need to make Integer.toString. You can simply pass the integer.